### ISSUE TITLE
Protection from bribing

### ISSUE THREAD

**Issue by xuhcc** [clrfund/monorepo#113]
Update contribution widget according to this description https://forum.clr.fund/t/making-contributions-to-a-funding-round/22/51

- Vote immediately after making contribution (requires #63).
- Publish key-change message before voting.

**Comment by xuhcc** [comment]
After recent MACI update, re-publishing message batch will probably be enough (#33).

**Comment by auryn-macmillan** [comment]
> After recent MACI update, re-publishing message batch will probably be enough (#33).

Do we want to give users away to broadcast the message from a separate account?

**Comment by xuhcc** [comment]
Yes. We should also think about relayers. But this will probably be rescheduled for a next milestone.

**Comment by xuhcc** [comment]
TODO: show signing key and state index to user so he can build transaction and submit it from another account
